test/unittest/decomposer/decomposer_test.go

```diff
@@ -36,7 +36,7 @@ func TestExecutePlaybook(t *testing.T) {
 		Type:          "ssh",
 		ID:            "action--test",
 		Name:          "ssh-tests",
-		StepVariables: cacao.Variables{expectedVariables.Name: expectedVariables},
+		StepVariables: cacao.NewVariables(expectedVariables),
```

Comment:
In test we should use the cacao.Variables{expectedVariables.Name: expectedVariables}, to have a different code path and prevent catching our own implementation mistakes

Reply:
I've added a variable test to make sure the NewVariables and Variables{"key": variableObject} is the same.